website/addons/base/views.py

```diff
@@ -223,6 +227,42 @@ def get_or_start_render(file_guid, start_render=True):
     return None
 
 
+def file_content(file_guid):
+    content = None
+    if is_editable(file_guid):
+        content = get_file_contents(
+            file_guid.mfr_download_url,
+            file_guid.mfr_cache_path,
+            file_guid.mfr_temp_path,
+            file_guid.public_download_url,
+        )
+
+    return content
+
+
+def is_editable(file_guid):
+    try:
+        file_guid.enrich()
+    except exceptions.AddonEnrichmentError:
+        return False
+
+    file_type = get_mimetype(file_guid.waterbutler_path)
+    if file_type is not None:
+        file_type = file_type.split('/')[0]
+        if file_type == 'text':
+            return True
+    else:
+        return False
+
+
+def ensure_path(path):
```

Comment:
```ensure_path``` is defined twice. Either create a utils.py file to contain shared renderer code, or import the definition from where it's first defined.